### PR TITLE
Insert to index

### PR DESCRIPTION
**Pull request by brainsiq**
Crude, simplistic implementation of inserting an object into an index. Getting this right and merging in should hopefully give us both the base files/model we need to be able to work on the rest of indexing and also searching, in parallel. The elasticsearch host obviously needs proper configuration at some point - that URL is for my Found free trial cluster.

### CODE REVIEW

**Comment by brainsiq** [comment]
Updated

**Comment by brainsiq** [comment]
Does this comment appear on slack?

**Comment by brainsiq** [comment]
Suckers!

File: test/index/insert_tests.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  describe('insert', function () {
+    let elasticStub;
+    let indexedObject;
+    let index;
+    let indexName;
+
+    beforeEach(function () {
+      elasticStub = sinon.stub(elasticsearch, 'Client', function () {
+        return {
+          index: function (args) {
```
Comment (toboid) [inline test/index/insert_tests.js]: Method syntax is available for object literals too if you want it: 

```javascript
return {
  index(args) {
    indexedObject = args.body;
    indexName = args.index;

    return new Promise(function (resolve) {
      resolve();
    });
  }
```

**Comment by toboid** [comment]
looks good :+1:

File: test/index/insert_tests.js (hunk 1 above)
Comment (brainsiq) [inline test/index/insert_tests.js]: Good spot, I updated that.

**Pull request merged**